Obsidian/WorldData/Generators/Overworld/ChunkBuilder.cs

```diff
@@ -109,16 +143,113 @@ internal static void Surface(GenHelper helper, Chunk chunk)
                         }
                         else
                         {
-                            chunk.SetBlock(x, y, z, BlocksRegistry.Stone);
+                            if (y <= 0)
+                            {
+                                chunk.SetBlock(x, y, z, BlocksRegistry.Deepslate);
+                            }
+                            else
+                            {
+                                chunk.SetBlock(x, y, z, BlocksRegistry.Stone);
+                            }
                         }
                     }
                 }
 
+                chunk.SetBlock(x, -64, z, BlocksRegistry.Bedrock);
                 chunk.Heightmaps[HeightmapType.WorldSurfaceWG].Set(x, z, terrainHeight);
             }
         }
     }
 
+    internal static bool GenerateOreCheck(int height, int OreType)
```

Comment:
Turn OreType into an enum instead of an int so the code can be discerned easier.

Reply:
This is my first time to write c# code so I don't know how to create an enum in c#, could you help me?
Thank you so much!

Reply:
I can, b/c it's my awful code you're working on :)

Enums are cool b/c they basically let you give a name to a number. Instead of `6`, we get to write `OreType.Emerald`. It makes code easier to read.
I would add this enum locally:
```csharp
    internal enum OreType : int
    {
        Coal,
        Iron,
        Copper,
        Gold,
        Lapis,
        Redstone,
        Emerald,
        Diamond
    }
```

Then we get to use it. C# has a new switch syntax that makes the GenerateOreCheck function super-concise.
Instead of saying `if(condition) {return true;} else {return false;}` we can just write the condition:
```csharp
    internal static bool GenerateOreCheck(int height, OreType oreType) => oreType switch
    {
// Read this as: if the ore type is coal, return whether the height is greater than 0 and less than 320
        OreType.Coal => height >= 0 && height <= 320,
        OreType.Iron => (height >= -63 && height <= 72) || (height >= 80 && height <= 320),
        OreType.Copper => height >= -16 && height <= 112,
        OreType.Gold => height >= -63 && height <= 30,
// and so on for the rest
        _ => true
    };
```

Then later, we can loop over the enum (instead of i being an arbitrary number which we have to go find the value for)... but we have to cast it back to an `int` in places where we actually want a number, which looks like `(int)i`:
```csharp
var orePlaced = false;
foreach (OreType i in Enum.GetValues(typeof(OreType)))
{
    // Check if we should be placing a given ore at this Y level
    if (!GenerateOreCheck(y, i))
    {
        // move onto the next ore
        continue;
    }

    var oreNoise1 = helper.Noise.Ore((int)i).GetValue(worldX, y, worldZ);
    var oreNoise2 = helper.Noise.Ore((int)i + ores.Length).GetValue(worldX, y, worldZ);
    if (oreNoise1 > 1.0 - OreSize && oreNoise2 > 1.0 - OreSize)
    {
        // Check that Emerald is only placed in the appropriate biomes
        var b = chunk.GetBiome(x, y, z);
        if (i == OreType.Emerald && !emeraldBiomes.Contains(b))
        {
            // no emerald in this biome
            // move onto the next ore
            continue;
        }

        // If Y is below 0, switch to deepore varients
        chunk.SetBlock(worldX, y, worldZ, y > 0 ? ores[(int)i] : deepores[(int)i]);
        orePlaced = true;
        break;
    }
}
if (orePlaced) { continue; }
```

Reply:
Thanks for the help!